components/authorization/backend/src/main/kotlin/service/AuthorizationService.kt

```diff
@@ -93,11 +93,11 @@ interface AuthorizationService {
     suspend fun listUsersWithRole(role: Role, compoundHierarchyId: CompoundHierarchyId): Set<String>
 
     /**
-     * Return a [Map] with the IDs of all users and their assigned roles on the hierarchy element identified by the
+     * Return a [Map] with the IDs of all users and their assigned role on the hierarchy element identified by the
      * given [compoundHierarchyId]. The result includes users who inherit access rights on this hierarchy element from
-     * higher levels, such as organization admins.
+     * higher levels, such as organization admins, but no superusers.
      */
-    suspend fun listUsers(compoundHierarchyId: CompoundHierarchyId): Map<String, Set<Role>>
+    suspend fun listUsers(compoundHierarchyId: CompoundHierarchyId): Map<String, Role>
```

Comment:
Do I understand this correctly that this function solves the issue that when you list the users for a repository, you also want to see the users who inherited the role from a parent entity? If so, I think it would be a great addition for later to include in the result if the role was inherited or not, this will be very useful for the UI.

Reply:
Yes, the returned map includes users who inherit their access rights from other levels. Adding this information to the result is possible, however, it would probably require some changes in the data structures used by the underlying `HierarchyPermission` class. I would therefore like to defer this change and first focus on the basics to get the new authorization component life.